### PR TITLE
use additionalProperties to create maps with defined key types

### LINKED ISSUE #377: CustomResourceDefinition validation fails for type map[string]string
**Describe the bug**
Applying a custom resource definition with type map[string]string for a field to a Kubernetes v1.20 cluster yields a validation error

`The CustomResourceDefinition "{mycrd.com}" is invalid: {long path into the crd}.type: Unsupported value: "map[string]string": supported values: "array", "boolean", "integer", "number", "object", "string"`

which makes `Dictionary<string, string>`  as a data type for a custom resource unusable.

**To Reproduce**
Define entities for a custom resource definition with the data type `Dictionary<string,string>` . (Try to) apply the CRD generated by KubeOps to your Kubernetes cluster.

**Expected behavior**
Validation should succeed and the CRD should be registered in the cluster.

### PR DESCRIPTION
**Pull request by hypnopotamus**
map[keyType]valueType is a valid resulting type, it is not a valid OpenAPI v3 type

fixes #377

### CODE REVIEW

**Comment by buehler** [comment]
oh nice! thank you :) this closes #383 as well right? I'd just revert the other one, but this is way better.

File: src/KubeOps/KubeOps.csproj [hunk 1]
```diff
@@ -21,6 +21,7 @@
             It enables a developer to create a custom controller for CRDs
             (CustomResourceDefinitions) that runs on kubernetes.
         </PackageDescription>
+        <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
```
Comment (buehler) [inline src/KubeOps/KubeOps.csproj:24]: That one is not needed ;-)

**Review by buehler — CHANGES_REQUESTED** [review]
Thank you for the fix! One little thing remains, the generate on build is not necessary since that is done via build script.

File: src/KubeOps/KubeOps.csproj (hunk 1 above)
Comment (hypnopotamus) [reply src/KubeOps/KubeOps.csproj]: gone now.  I turned that on for testing (reference the output directory as a nuget feed from another project), left behind accidentally when committing

**Pull request merged**